### DISCUSSION TITLE
Pandoc 3.1.6 fails on creating a table, where 2.7.3 works

### DISCUSSION THREAD

**Discussion by sparlund** [jgm/pandoc#9533] [Q&A] [answered]
Hi,

I use pandoc to generate PDF's from Markdown, using the lualatex engine. I use a bunch of filters, extensions and latex packages to get the PDF looking like I want it. I'm trying to upgrade to Pandoc version 3.1.6, and then my PDF generation fails on creating a table as below
````
+-------------------------+-------------------------------+----------------------------+
| **A**                   | **B**                         | **C**                      |
+=========================+===============================+============================+
| D                       | E                             | F                          |
+-------------------------+-------------------------------+----------------------------+
| G                       | H                             | I                          |
+-------------------------+-------------------------------+----------------------------+
````
But I get this error:
````
! Missing number, treated as zero.
<to be read again> 
(
l.409 \begin
          {minipage}[b]{\linewidth}\raggedright
````
If I use Pandoc 2.7.3 this table is generated as below:
````
\begin{tablenos:no-prefix-table-caption}

\begin{longtable}[]{@{}lll@{}}
\toprule
\begin{minipage}[b]{0.27\columnwidth}\raggedright
\textbf{A}\strut
\end{minipage} & \begin{minipage}[b]{0.34\columnwidth}\raggedright
\textbf{B}\strut
\end{minipage} & \begin{minipage}[b]{0.30\columnwidth}\raggedright
\textbf{C}\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.27\columnwidth}\raggedright
D\strut
\end{minipage} & \begin{minipage}[t]{0.34\columnwidth}\raggedright
E\strut
\end{minipage} & \begin{minipage}[t]{0.30\columnwidth}\raggedright
F\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.27\columnwidth}\raggedright
G\strut
\end{minipage} & \begin{minipage}[t]{0.34\columnwidth}\raggedright
H\strut
\end{minipage} & \begin{minipage}[t]{0.30\columnwidth}\raggedright
I\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}
\end{tablenos:no-prefix-table-caption}
````
Here are is my Pandoc command:

````
pandoc --toc --listings --template=latex_template.tex \
				--number-sections --pdf-engine=lualatex \
				-s\
				--verbose\
				-V block-headings \
				--from markdown+multiline_tables+grid_tables+backtick_code_blocks+yaml_metadata_block\
				--filter pandoc-tablenos -M tablenos-warning-level=0\
				--filter pandoc-fignos -M fignos-warning-level=0\
				--filter pandoc-plantuml\
				-o temp.pdf temp.md
````
I have different paths for pandoc 2.7.3 and 3.1.6 and when I switch between I can generate this error.

Here are some latex packages I use:
````
\usepackage{longtable,booktabs}
\usepackage{caption}
````

What could be the issue? I see what the different pandoc versions generate different latex code.

**Comment by jgm** [comment]
It works fine with the latest version 3.1.12.2 -- upgrade?

**Reply by jgm** [reply]
PS. I did not test with all the same options as you  -- just with a simple `pandoc -o test.pdf`.
If things don't work with 3.1.12.2, then you'd need to investigate the possibility that one of your filters is creating the problem.  (Do that by removing them one by one.)

**Reply by sparlund** [reply]
Thanks for the input, now that you're saying the obvious approach is to remove one flag at a time.

This led me to identifying that the issue comes from the latex template. If I remove the table from `temp.md` I'm still able to generate the PDF. The template includes a table in the header, which I create using `tabularx`.

I tried removing the extensions on `--from markdown+multiline_tables+grid_tables+backtick_code_blocks+yaml_metadata_block` to just `--from markdown` but it had no affect.

I'm confused by the different ways to generate tables in Markdown, I can't remember exactly how I landed on this approach with grid around all elements, and why I have every specific option on `--from markdown`.

Does any of the extensions there stick out to you?

Could the difference be that `\linewidth` is not a OK value for `\begin{minipage}`?
2.7.3: `\begin{minipage}[b]{\linewidth}\raggedright` NOK
3.1.6: `\begin{minipage}[b]{0.27\columnwidth}\raggedright`

> It works fine with the latest version 3.1.12.2 -- upgrade?

Regarding updating. I'm on enterprise Linux and kind of locked on what versions I can use, do you think it would help to upgrade still in this scenario? Then I'd look into it.

**Reply by jgm** [reply]
`--from markdown+multiline_tables+grid_tables+backtick_code_blocks+yaml_metadata_block` is the same as `--from markdown` because those options are enabled by default.

**Comment by sparlund** [comment, marked as answer]
After some more googling I realized the issue was the same as in this discussion:
[https://github.com/jgm/pandoc/issues/8248](https://github.com/jgm/pandoc/issues/8248)

The issue is the missing `\usepackage{calc} % for calculating minipage widths`.

Thanks for the support @jgm :) This is a latex "issue" and nothing to do with Pandoc.

**Reply by andylarkum** [reply]
OH MY GOODNESS!!! Thank you so much, I have been trying to solve this for....far too long!

**Comment by miasma** [comment]
Had this same issue with pandoc 3.1.11.1-17 on Arch. Thanks for solving it. Hm I thought Arch was a bleeding edge distro, but apparently the pandoc package is pretty ancient.

**Reply by tarleb** [reply]
It's generally better to use [pandoc-bin](https://aur.archlinux.org/packages/pandoc-bin) from AUR, as it's well maintained and doesn't pull in as many Haskell dependencies.
